Dockerfile

```diff
@@ -14,15 +14,39 @@ RUN rm src/lib.rs
 COPY src/* src/
 # build the executable
 RUN cargo build --release
+# download jq
+RUN mkdir -p ${wd}/bin
+RUN curl -L -o /usr/local/bin/jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 \
+  && chmod +x /usr/local/bin/jq
+# download cargo-local-registry (we don't use cargo install because we want it portable: we'll copy it later to the image that is run)
+RUN curl -L -o clr.tar.gz https://github.com/ChrisGreenaway/cargo-local-registry/releases/download/0.2.1/cargo-local-registry-0.2.1-x86_64-unknown-linux-musl.tar.gz \
+  && tar xvzf clr.tar.gz && chmod +x cargo-local-registry && mv cargo-local-registry /usr/local/cargo/bin
+# download and build popular crates to local registry
+RUN mkdir /local-registry
+WORKDIR /local-registry
+COPY local-registry/* ./
+RUN curl "https://crates.io/api/v1/crates?page=1&per_page=100&sort=downloads" | \
+    jq -r '.crates[] | .id + "=\"" + .max_stable_version + "\""' >> Cargo.toml
```

Comment:
How big are crates? Is it worth doing 1000 instead of 100? Or are they big?

Reply:
For the top 100 crates it's just 17MB

Reply:
I'd be up for expanding it to the top 500 then? @ErikSchierboom?

Reply:
🤷 It all depends on how likely it is those crates are used. I have no data to back up any claims though. @coriolinus @dhovart how likely is it that students would used a crate that is not in the top 100 crates?

Reply:
I have absolutely no idea, sorry! I just know that this list should maybe be fine-tuned. There are crates we pull that make no sense in the context of exercism, such as gui development.
Maybe there should be a message somewhere (below the list of supported crates maybe) that say that if they want support for a given crate, they should open a GH issue?

Reply:
Instead of the crates.io API, we would get the list from somewhere else, where it was edited.
I guess I can add an endpoint to exercism's site API. But I'm not familiar with the site codebase yet, is some administration interface where someone could create this list ?

Reply:
> I have absolutely no idea, sorry! I just know that this list should maybe be fine-tuned. 

I think having a curated list makes the most sense. We definitely don't need to have any GUI crates included, and that would also allow us to cherry-pick crates that are not in the top 100, but still useful for students in the context of Exercism.

> Instead of the crates.io API, we would get the list from somewhere else, where it was edited.

We haven't yet come up with a spec for this. I think we want this list to be in the track repo at [exercism/rust](https://github.com/exercism/rust). Having it there means that we'll be able to programmatically access it from the website. The next question then becomes: where to store it? I think there are basically two options:

1. Add a section to the root `config.json` file
2. Add a new file just to list the supported libraries

I don't yet have a real preference, but the first option _might_ be best.

Reply:
I think it would be fine as MVP to get it just as a file in this repo that's read when Docker builds. That should be quite trivial. Then we can improve it later. In my eyes this is a real priority, so the quickest solution is probably the best one! :)

Reply:
> I think it would be fine as MVP to get it just as a file in this repo that's read when Docker builds. 

I wouldn't want to touch the existing Docker push workflow (we want to sync that across all repos at some point) and secondly, I'd prefer it to be in exercism/rust as there will be more visibility with maintainers (and students will figure out that location sooner than they will this repo).